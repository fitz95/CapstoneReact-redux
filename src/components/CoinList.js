import React from 'react';
import { useSelector } from 'react-redux';
import CoinUi from './CoinUi';
import '../css/coinlist.css';

function CoinList() {
  const coins = useSelector((store) => store.coins.list);
  return (
    <div className="coin-div">
      <div className="coinsheading">
        <h2>Crypto Market Prices</h2>
      </div>
      <div className="coinsdiv">
        {coins.map((coins) => (
          <CoinUi
            id={coins.id}
            key={coins.id}
            name={coins.name}
            picture={coins.image}
            currentPrice={coins.current_price}
          />
        ))}
      </div>
    </div>
  );
}

export default CoinList;
