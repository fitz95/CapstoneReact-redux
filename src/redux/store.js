import { configureStore, combineReducers } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import coinsReducer from './coins/coinsSlice';
import coinDetailsReducer from './coindetails/coinDetailsSlice';

const store = configureStore({
  reducer: combineReducers({
    coins: coinsReducer,
    coinDetails: coinDetailsReducer,
  }),
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
