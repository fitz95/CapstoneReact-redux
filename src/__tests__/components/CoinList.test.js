import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from 'src/redux/store';
import CoinList from 'src/components/CoinList';

describe('RocketsLists', () => {
  it('renders correctly for reserved state', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <CoinList />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
