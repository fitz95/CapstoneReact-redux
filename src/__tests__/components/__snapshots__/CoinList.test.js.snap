// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketsLists renders correctly for reserved state 1`] = `
<div
  className="coin-div"
>
  <div
    className="coinsheading"
  >
    <h2>
      Crypto Market Prices
    </h2>
  </div>
  <div
    className="coinsdiv"
  />
</div>
`;
